options/internal/generic/locale.cpp

```diff
@@ -1,98 +1,627 @@
+	frg::span<const uint32_t> offsets{reinterpret_cast<const uint32_t *>(rec.data() + 8), elements};
+
+	switch(category) {
+		case LC_CTYPE: {
+			parse_category_array(out->ctype.members, ctype_parser, rec, offsets);
+			break;
+		}
+		case LC_NUMERIC: {
+			parse_category_array(out->numeric.members, numeric_parser, rec, offsets);
+			break;
+		}
+		case LC_TIME: {
+			parse_category_array(out->time.members, time_parser, rec, offsets);
+			break;
+		}
+		case LC_COLLATE:
+			parse_category_array(out->collate.members, collate_parser, rec, offsets);
+			break;
+		case LC_MONETARY: {
+			parse_category_array(out->monetary.members, monetary_parser, rec, offsets);
+			break;
+		}
+		case LC_MESSAGES: {
+			parse_category_array(out->messages.members, messages_parser, rec, offsets);
+			break;
+		}
+		// skip LC_ALL
+		case LC_PAPER: {
+			parse_category_array(out->paper.members, paper_parser, rec, offsets);
+			break;
+		}
+		case LC_NAME: {
+			parse_category_array(out->name.members, name_parser, rec, offsets);
+			break;
+		}
+		case LC_ADDRESS: {
+			parse_category_array(out->address.members, address_parser, rec, offsets);
+			break;
+		}
+		case LC_TELEPHONE: {
+			parse_category_array(out->telephone.members, telephone_parser, rec, offsets);
+			break;
+		}
+		case LC_MEASUREMENT: {
+			parse_category_array(out->measurement.members, measurement_parser, rec, offsets);
+			break;
+		}
+		case LC_IDENTIFICATION: {
+			parse_category_array(out->identification.members, identification_parser, rec, offsets);
+			break;
+		}
+		default:
+			mlibc::infoLogger() << "mlibc: category " << lcNames[category] << " unknown!" << frg::endlog;
+			return false;
+	}
+
+	return true;
+}
+
+bool findLocaleFileRecord(int category, const char *name, mlibc::localeinfo *out) {
+	auto parsedName = parseLocaleName(name);
+	auto list = buildLocaleList(parsedName);
+
+	for (auto lp : locpaths) {
+		for (auto locale : list) {
+			frg::string<MemoryAllocator> path{lp, getAllocator()};
+			path += '/';
+			path += locale;
+			path += '/';
+			if (category == LC_MESSAGES)
+				path += "LC_MESSAGES/SYS_LC_MESSAGES";
+			else
+				path += lcNames[category];
+
+			auto e = localeFiles.find(path);
+			if (e != localeFiles.end()) {
+				auto &[name, window] = *e;
+				if (parseCategoryInfo(category, {static_cast<const uint8_t *>(window->get()), window->size()}, out))
+					return true;
+			} else {
+				struct stat info;
+				if (mlibc::sys_stat(mlibc::fsfd_target::path, -1, path.data(), 0, &info))
+					continue;
+
+				auto window = smarter::allocate_shared<file_window>(getAllocator(), path.data());
+				localeFiles.insert(path, window);
+
+				if (parseCategoryInfo(category, {static_cast<const uint8_t *>(window->get()), window->size()}, out)) {
+					return true;
+				}
+			}
+		}
+	}
+
+	return false;
+}
+
+bool findLocaleArchiveRecord(int category, const char *name, mlibc::localeinfo *out) {
+	if (!localeArchive)
+		return false;
+
+	__ensure(category != LC_ALL);
```

Comment:
Can't we just return false here? I know we're AssertOS but if we can gracefully fail here I think that's better.